EIPS/eip-3675.md

```diff
@@ -191,6 +197,13 @@ This event was removed for two reasons:
 1. This event was an unnecessary optimization to allow for pruning of "bad" blocks from the block tree. This optimization was unnecessary because the PoS consensus would never send `POS_FORKCHOICE_UPDATED` for any such bad blocks or their descendants, and eventually any such blocks would be able to be pruned after a PoS finality event of an alternative branch in the block tree.
 2. This event was dangerous in some scenarios because a block could be referenced by two *different* and conflicting PoS branches. Thus for the same block in some scenarios, both a `POS_CONSENSUS_VALIDATED == TRUE` and `POS_CONSENSUS_VALIDATED == FALSE` event could sent, entirely negating the ability to safely perform the optimization in (1).
 
+### EIP-2124 fork identifier
+
+The value of `FORK_NEXT` refers to the block number of the next fork a given node knows about and `0` otherwise.
+As the block number of the terminal PoW block will not be known until the transition, nodes have no value to use for `FORK_NEXT` and in light of this fact will use the default `0`.
+Nodes can retroactively update their node identifier with the correct data once the fork block is known.
```

Comment:
Hmm, I think the retroactive update process should be clearly specified.

Does it mean that at `TRANSITION_POS_BLOCK` the `FORK_HASH` changes to include `TRANSITION_POS_BLOCK`?

Reply:
right, once the transition has happened, `TRANSITION_POS_BLOCK` will have a block number `N`, e.g. in `TRANSITION_POS_BLOCK.body.execution_payload.block_number`.

we don't know `N` ahead of time as we use the more dynamic total difficulty trigger condition.

once we do know `N`, nodes can set `FORK_HASH` to something like:

```
CRC32(<genesis-hash> || uint64(1150000) || uint64(1920000) || ... <intervening fork block numbers> ... || uint64(N))
```

`FORK_NEXT` will remain `0` until we know the block number of the next fork (or decide otherwise how to handle the EIP-2124 scheme in a post-merge world).

Reply:
I think it would be beneficial explaining this in the EIP, next to where it states that it should be 0 initially.